server/functions/binary/concatenate.go

```diff
@@ -51,6 +52,19 @@ var anytextcat = framework.Function2{
 	},
 }
 
+// byteacat represents the PostgreSQL function of the same name, taking the same parameters.
+var byteacat = framework.Function2{
+	Name:       "byteacat",
+	Return:     pgtypes.Bytea,
+	Parameters: [2]pgtypes.DoltgresType{pgtypes.Bytea, pgtypes.Bytea},
+	Strict:     true,
+	Callable: func(ctx *sql.Context, paramsAndReturn [3]pgtypes.DoltgresType, val1 any, val2 any) (any, error) {
+		v1 := val1.([]byte)
+		v2 := val2.([]byte)
+		return append(v1, v2...), nil
```

Comment:
`append` will write to the underlying array on `v1`. If `v1` shares its underlying array with another slice, then we may overwrite on the original slice.
https://go.dev/play/p/Fwf_920pVNT
It's better to `copy` here.